### PR TITLE
docs: treatment of badges in the sidebar depending on development mode

### LINKED ISSUE #2198: Update build_home() docs
With what was changed in #2114

What happens to the badges by default for production and for development websites. What can one customize or not (the presence of the dev section in the sidebar, the removal of badges from the main div)

### PR DESCRIPTION
**Pull request by maelle**
Fix #2198

### CODE REVIEW

**Comment by maelle** [comment]
@hadley this documents the current behaviour and would fix #2198 now that I opened other more specific issues (#2239 + #2240 )

File: man/rmd-fragments/sidebar-configuration.Rmd [hunk 1]
```diff
@@ -31,7 +31,7 @@ These are drawn from seven built-in components:
 
 -   `authors`: selected authors from the `DESCRIPTION`.
 
--   `dev`: development status badges found in `README.md`/`index.md`.
+-   `dev`: development status badges found in `README.md`/`index.md`. *This section is automatically removed for websites of released versions (see "Development mode" in `?build_site`. For all development modes, the badges are removed from the main content.*
```
Comment (hadley) [inline man/rmd-fragments/sidebar-configuration.Rmd:34]: Rather than saying when it's removed, might be clearer to state when it's shown?

Comment (maelle) [reply man/rmd-fragments/sidebar-configuration.Rmd]: Is the new phrasing any better? See also #2240: if we add a parameter, then the docs will need to change again :zany_face:

File: man/rmd-fragments/sidebar-configuration.Rmd [hunk 2]
```diff
@@ -31,7 +31,7 @@ These are drawn from seven built-in components:
 
 -   `authors`: selected authors from the `DESCRIPTION`.
 
--   `dev`: development status badges found in `README.md`/`index.md`. *This section is automatically removed for websites of released versions (see "Development mode" in `?build_site`. For all development modes, the badges are removed from the main content.*
+-   `dev`: development status badges found in `README.md`/`index.md`. *This section is automatically removed for websites of released versions (For all development modes, the badges are removed from the main content. See "Development mode" in `?build_site` for a summary of when badges are shown in the sidebar as opposed to not shown anywhere on the home page)*
```
Comment (hadley) [inline man/rmd-fragments/sidebar-configuration.Rmd:34]:
```suggestion
-   `dev`: development status badges extracted from `README.md`/`index.md`. This is only shown for "development" versions of websites; see "Development mode" in `?build_site` for details. 
```

**Comment by maelle** [comment]
I still feel there should be an explicit mention somewhere of the _removal_ of badges from the main content (even if that's what "extract" means, it could be even clearer). However it will happen with #2240 anyway so happy to merge this one if you're ok with it.

**Comment by hadley** [comment]
Maybe it's worth also updating https://pkgdown.r-lib.org/reference/build_home.html#dev-badges to mention that the badges are removed from the home page and put in the sidebar?

**Comment by maelle** [comment]
Ah yes excellent place for this!

**Comment by maelle** [comment]
Ok to merge?

**Review by hadley — APPROVED** [review]
Sorry, yes, forgot to approve it earlier.

**Pull request merged**